UI updates & parse script

alaaltoros opened the pull request:
PR related to issue #21 (Front-End UI for MVP)

It includes the following:

Basic UI
Login/logout with google
Location data parse script
I Will be updating it as soon as other parts are ready.

P.S. The login/logout functionality requires google client id (check the README.md) and the upload button will output the parsed data in the console.

alaaltoros committed:
Google login

alaaltoros committed:
Contribute page component skeleton

alaaltoros committed:
Contribute page steps

alaaltoros committed:
Contribute, Auth & page 3

alaaltoros committed:
Parse script updated

ainsleys commented:
Hey @alaaltoros can you add a description to this PR explaining what it is?

alaaltoros commented:
PR related to [issue #21 (Front-End UI for MVP)](https://github.com/enigmampc/SafeTrace/issues/21)

It includes the following:
- Basic UI
- Login/logout with google
- Location data parse script

I Will be updating it as soon as other parts are ready.

P.S. The login/logout functionality requires google client id (check the README.md) and the upload button will output the parsed data in the console.

alaaltoros committed:
Integrate login with API

alaaltoros committed:
Rrport & Location history upload

alaaltoros commented:
Added report and file upload basic functionality

lacabra commented on UIcode/README.md:
```diff
@@ -0,0 +1,25 @@
+echo "REACT_APP_API_URL=http://localhost:4080" >> .env.development
+
+#set google client id i.e. 119469794689-hhq7rpcmd88c7r5gkiom0u2pakfka3cd.apps.googleusercontent.com
+echo "REACT_APP_GOOGLE_CLIENT_ID=119469794689-hhq7rpcmd88c7r5gkiom0u2pakfka3cd.apps.googleusercontent.com" >> .env.development
```
Can you please document from where one gets the Google Client ID, ideally with some instructions on how to obtain that ID? And how to configure the Origin/Authorized domains so that this application works?

assafmo replied:
@alaaltoros ?

lacabra commented on UIcode/README.md:
```diff
@@ -0,0 +1,25 @@
+touch .env.development
+
+# set API url i.e. http://localhost:4080
+echo "REACT_APP_API_URL=http://localhost:4080" >> .env.development
```
I am confused by this instruction to configure the API on port 4080, but the server is started below on port 3000. Can you please document this better, explaining the difference between the two ports if they are indeed different ports?

assafmo replied:
@alaaltoros ?

alaaltoros replied:
This is the URL of the backend API, I added more description to this file.

lacabra commented on client/scripts/last_14d_visits.js:
```diff
@@ -1,40 +0,0 @@
-<script type="text/javascript">
```
Have you deleted this file by mistake? If not, can you please make an argument on why you are deleting this file?

alaaltoros replied:
This script was replaced by `UIcode/client/src/Services/parser.js` module

lacabra commented on UIcode/server.js:
```diff
@@ -0,0 +1,26 @@
+const express = require('express');
```
There is no mention on how to start this server on the README. Can you please add documentation there and how it relates with the client?

alaaltoros replied:
This is not needed anymore, I will delete it

lacabra reviewed (changes requested):
Dear @alaaltoros, thanks for your contribution in building a UI for this project. I have done an initial review of this PR, but I cannot get it to work, meaning: I have installed the package deps, configured the client, and launched it, and it displays on the browser. The connection with Google OAuth seems fine, but after trying to log in, I get an error message `Login failed`. I believe this may be due mostly to gaps in documentation. Kindly review the items I've highlighted in my review, and ping me with a mention when these are addressed, so that I can do a more in-depth review.

Thanks again!

alaaltoros committed:
Remove UIcode/server.js and update directory structure

alaaltoros commented:
I updated the `README.md` file and the directory structure.
Let me know if you can get the client to run or if you have any other comments or doubts.
Please note that the backend is included in a different repository and a different pull request (I added a URL in the `README.md` file)

alaaltoros committed:
Results formatting

alaaltoros commented:
Added the Results page with formatted data. However, I could not test with a real response from **findMatches** endpoint of Enclave; I received no matches and could not upload report new results because of the CORS error.

alaaltoros committed:
Correct README.md

cmalfesi commented:
@alaaltoros I have tested your last changes and in general, it looks good. Regarding the issue that returns CORS error, I have done some tests and I see that the problem occurs when there are 2 consecutive calls to the Enclave endpoints. If I execute one only it works. I think the problem is related to the promise or await functions.

cmalfesi commented:
@alaaltoros 
I have an improvement for the frontend to get better behavior with consecutive calls to the enclave endpoints. We need to separate `addPersonalData` and `findMatch` in different async functions. This solves the CORS problem. I'll send you the required changes.

alaaltoros committed:
Updating enclave.js

assafmo commented:
@alaaltoros can you resolve the conflicts? Also waiting for @cmalfesi CORS fixes in order to merge this.

alaaltoros commented:
@assafmo conflicts resolved!

assafmo commented:
Still seeing conflicts in `UIcode/client/package-lock.json` and `UIcode/client/yarn.lock`.
You can remove the one you don't use and rebuild the one you use.

lacabra committed:
MAINT: remove conflicting yarn.lock
